xbmc/guilib/GUIFontTTF.cpp

```diff
@@ -390,6 +389,7 @@ void CGUIFontTTF::DrawTextInternal(CGraphicContext& context,
   if (dirtyCache)
   {
     // save the origin, which is scaled separately
+    std::vector<Glyph> glyphs = GetHarfBuzzShapedGlyphs(text);
```

Comment:
While you are at it, make it `const std::vector`.

```
diff --git a/xbmc/guilib/GUIFontTTF.cpp b/xbmc/guilib/GUIFontTTF.cpp
index 285dcdbdac..4038407641 100644
--- a/xbmc/guilib/GUIFontTTF.cpp
+++ b/xbmc/guilib/GUIFontTTF.cpp
@@ -363,7 +363,6 @@ void CGUIFontTTF::DrawTextInternal(CGraphicContext& context,
   }
 
   Begin();
-  std::vector<Glyph> glyphs = GetHarfBuzzShapedGlyphs(text);
   uint32_t rawAlignment = alignment;
   bool dirtyCache(false);
   bool hardwareClipping = m_renderSystem->ScissorsCanEffectClipping();
@@ -390,6 +389,7 @@ void CGUIFontTTF::DrawTextInternal(CGraphicContext& context,
   if (dirtyCache)
   {
     // save the origin, which is scaled separately
+    const std::vector<Glyph> glyphs = GetHarfBuzzShapedGlyphs(text);
     m_originX = x;
     m_originY = y;
 
@@ -580,7 +580,7 @@ float CGUIFontTTF::GetTextWidthInternal(const vecText& text)
 }
 
 // this routine assumes a single line (i.e. it was called from GUITextLayout)
-float CGUIFontTTF::GetTextWidthInternal(const vecText& text, std::vector<Glyph>& glyphs)
+float CGUIFontTTF::GetTextWidthInternal(const vecText& text, const std::vector<Glyph>& glyphs)
 {
   float width = 0;
   for (auto it = glyphs.begin(); it != glyphs.end(); it++)
diff --git a/xbmc/guilib/GUIFontTTF.h b/xbmc/guilib/GUIFontTTF.h
index 130c158cfb..23b6afb9f5 100644
--- a/xbmc/guilib/GUIFontTTF.h
+++ b/xbmc/guilib/GUIFontTTF.h
@@ -151,7 +151,7 @@ protected:
   std::vector<Glyph> GetHarfBuzzShapedGlyphs(const vecText& text);
 
   float GetTextWidthInternal(const vecText& text);
-  float GetTextWidthInternal(const vecText& text, std::vector<Glyph>& glyph);
+  float GetTextWidthInternal(const vecText& text, const std::vector<Glyph>& glyph);
   float GetCharWidthInternal(character_t ch);
   float GetTextHeight(float lineSpacing, int numLines) const;
   float GetTextBaseLine() const { return static_cast<float>(m_cellBaseLine); }
```

Reply:
Done.